src/flag_gems/ops/slice_scatter.py

```diff
@@ -2,584 +2,39 @@
-    out = tl.where(b_row_mask & col_mask, src, inp)
-    tl.store(A_out + a_offset, out, mask=a_row_mask & col_mask)
-
-
-def scatter_2d_inner(x, y, x_out, start, end, step):
-    nrow, x_ncol = x_out.size()
-    y_ncol = y.size(1)
-    x_stride_i, x_stride_j = x_out.stride()
-    y_stride_i, y_stride_j = y.stride()
-    strided = step > 1
-    pivoted = y_stride_i < y_stride_j
-    grid = lambda meta: (
-        triton.cdiv(nrow, meta["R"]),
-        triton.cdiv(x_ncol, meta["C"]),
-    )
-    scatter_2d_inner_kernel[grid](
-        x,
-        y,
-        x_out,
-        nrow,
-        x_ncol,
-        y_ncol,
-        x_stride_i,
-        x_stride_j,
-        y_stride_i,
-        y_stride_j,
-        start,
-        end,
-        step,
-        strided=strided,
-        pivoted=pivoted,
-    )
-
-
-def scatter_2d_outer(A, B, A_out, start, end, step):
-    nrow_A, ncol = A_out.size()
-    nrow_B = B.size(0)
-    a_stride0, a_stride1 = A_out.stride()
-    b_stride0, b_stride1 = B.stride()
-    grid = lambda meta: (
-        triton.cdiv(nrow_A, meta["NROW"]),
-        triton.cdiv(ncol, meta["NCOL"]),
-    )
-    scatter_2d_outer_kernel[grid](
-        A,
-        B,
-        A_out,
-        nrow_A,
-        nrow_B,
-        ncol,
-        a_stride0,
-        a_stride1,
-        b_stride0,
-        b_stride1,
-        start,
-        end,
-        step,
-        NROW=64,
-        NCOL=64,
-    )
-
-
-def scatter_3d_mid(A, B, A_out, start, end, step):
-    M, N1, K = A_out.size()
-    N2 = B.size(1)
-    am_stride, an_stride, ak_stride = A_out.stride()
-    bm_stride, bn_stride, bk_stride = B.stride()
-
-    grid = lambda meta: (M, triton.cdiv(N1, meta["NBLOCK"]))
-    scatter_3d_mid_kernel[grid](
-        A,
-        B,
-        A_out,
-        N1,
-        N2,
-        K,
-        am_stride,
-        an_stride,
-        ak_stride,
-        bm_stride,
-        bn_stride,
-        bk_stride,
-        start,
-        end,
-        step,
-        NBLOCK=512,
-        KBLOCK=triton.next_power_of_2(K),
-    )
-
-
-def slice_scatter_v2(inp, src, dim=0, start=None, end=None, step=1):
-    logging.debug("GEMS SLICE_SCATTER")
-    assert src.device == inp.device, "inp and src reside on different devices."
-    assert dim >= -inp.ndim and dim < inp.ndim, "Invalid dim"
-    assert step > 0, "slice step must be positive"
-    dim = dim % inp.ndim
-
-    start = start or 0
-    end = end or inp.size(dim)
-
     out = torch.empty_strided(
         inp.size(), inp.stride(), dtype=inp.dtype, device=inp.device
     )
 
-    # Look for a permute of dims so that the outer dims and inner dims relative to dim
-    # after permute can be coalesced.
-    # But this can be difficult so we're resorting to a suffcient and not necessary condition.
-    new_out, ordered_dims = simplify(out, dim)
-    if new_out is not None:
-        new_src, _ = simplify(src, dim, ordered_dims)
-    else:
-        new_src = None
+    ndim = inp.ndim
+    copy_.instantiate(ndim)

```

Comment:
Also, if a rank-specific overload is intended to be use, it has to be returned and used.